cogs/puzzle.py

```diff
@@ -22,10 +23,49 @@
 HUNT_START_TIME: datetime = config["HUNT_START_TIME"]
 
 
+class PaginationView(View):
+    def __init__(self, pages):
+        super().__init__()
+
+        self.page_num = 0
+        self.pages = pages
+
+    @button(
+        label="<",
+        style=discord.ButtonStyle.blurple,
+    )
+    async def prev_button(self, interaction: discord.Interaction, button: Button):
+        await interaction.response.defer()
+
+        if self.page_num > 0:
+            self.page_num += -1
+
+        await interaction.followup.edit_message(
+            message_id=interaction.message.id, embed=self.pages[self.page_num]
+        )
+
+    @button(
+        label=">",
+        style=discord.ButtonStyle.blurple,
+    )
+    async def next_button(self, interaction: discord.Interaction, button: Button):
+        await interaction.response.defer()
+
+        if self.page_num < len(self.pages) - 1:
+            self.page_num += 1
+
+        await interaction.followup.edit_message(
+            message_id=interaction.message.id, embed=self.pages[self.page_num]
+        )
+
+
 class Puzzle(commands.GroupCog):
     def __init__(self, bot):
         self.bot = bot
 
+    def update_leaderboard(self, new_embeds: list[discord.Embed]):
+        self.leaderboard_embeds = new_embeds
+
```

Comment:
I can't find where this function is used

Reply:
forgot to remove that